src/main/java/edu/ea/project/team8/domain/CourseOffering.java

```diff
@@ -5,46 +5,63 @@
 
 import javax.persistence.*;
 
+import lombok.NoArgsConstructor;
 import org.hibernate.annotations.Fetch;
 import org.hibernate.annotations.FetchMode;
 
 import lombok.Data;
 
 @Data
+@NoArgsConstructor
 @Entity
 public class CourseOffering {
 
-    @Id
-    @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private Integer id;
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Integer id;
 
-    @ManyToOne
-    @JoinColumn(name = "course_id", nullable = false)
-    private Course course;
+	@ManyToOne
+	@JoinColumn(name = "course_id", nullable = false)
+	private Course course;
 
-    @Column(columnDefinition = "nvarchar(50)")
-    private String period;
+	@Column(columnDefinition = "nvarchar(50)")
+	private String period;
 
-    @Column(nullable = false)
-    private LocalDate beginDate;
+	@Column(nullable = false)
+	private LocalDate beginDate;
 
-    @Column(nullable = false)
-    private LocalDate endDate;
+	@Column(nullable = false)
+	private LocalDate endDate;
 
-    @OneToMany
-    @JoinColumn(name = "offerid")
-    @Fetch(FetchMode.SUBSELECT)
```

Comment:
Why is this SUBSELECT?

Reply:
@JaCkIsO I didn't introduce that. But I think it is to load the collection lazily to prevent that n + 1 select problem.
But @albahrawy could explain it better.

Reply:
Makes sense